File: exercises/day15-No-Exceptions-For-Failures/kotlin/src/main/kotlin/santaChristmasList/operations/dependencies/Dependencies.kt
package santaChristmasList.operations.dependencies

import santaChristmasList.operations.models.Child
import santaChristmasList.operations.models.Gift

interface WishList {
    fun identifyGift(child: Child): Gift?
}

interface Factory {
    fun findManufacturedGift(gift: Gift): Gift?
}

interface Inventory {
    fun pickUpGift(barCode: String): Gift?
}

File: exercises/day15-No-Exceptions-For-Failures/scala/src/main/scala/santaChristmasList/operations/dependencies/Dependencies.scala
package santaChristmasList.operations.dependencies

import santaChristmasList.operations.models._

trait WishList {
  def identifyGift(child: Child): Gift
}

trait Factory {
  def findManufacturedGift(gift: Gift): Gift
}

trait Inventory {
  def pickUpGift(barCode: String): Gift
}
